anchor/programs/pubkey-protocol/src/state/identity.rs

```diff
@@ -59,6 +41,52 @@ impl Identity {
             PubkeyProfileError::InvalidProviderName
         );
 
+        // Validate provider_id based on the provider
+        match &self.provider {
+            PubKeyIdentityProvider::Solana => {
+                require!(
+                    Pubkey::from_str(&self.provider_id).is_ok(),
+                    PubkeyProfileError::InvalidSolanaPubKey
+                );
+            }
+            PubKeyIdentityProvider::Discord => {
```

Comment:
For all of these social options, we want to check for a valid provider id - which is not the url.

In the future we might want to add an `is_valid_provider_id` method, where we check if the id matches the expected format. 

For now though, let's make sure they are a string and store it that way - while we figure out the shape of the actual types.